Sources/Vitamin/Foundations/Typography/TextStyles.swift

```diff
@@ -88,124 +163,51 @@ extension VitaminTextStyle {
-
-        case .title3:
-            return baseAttributes(
-                sizeProvider: 20,
-                font: VitaminFontFamily.Roboto.bold,
-                lineHeight: 24
-            )
-
-        case .headline:
-            return baseAttributes(
-                sizeProvider: 16,
-                font: VitaminFontFamily.Roboto.bold,
-                lineHeight: 24
-            )
-
-        case .body:
-            return baseAttributes(
-                sizeProvider: 17,
-                font: VitaminFontFamily.Roboto.regular,
-                lineHeight: 28
-            )
-
-        case .callout:
-            return baseAttributes(
-                sizeProvider: 16,
-                font: VitaminFontFamily.Roboto.regular,
-                lineHeight: 24
-            )
-
-        case .subhead:
-            return baseAttributes(
-                sizeProvider: 15,
-                font: VitaminFontFamily.Roboto.regular,
-                lineHeight: 20
-            )
-
-        case .footnote:
-            return baseAttributes(
-                sizeProvider: 14,
-                font: VitaminFontFamily.Roboto.regular,
-                lineHeight: 20
-            )
-
-        case .caption1:
-            return baseAttributes(
-                sizeProvider: 12,
-                font: VitaminFontFamily.Roboto.regular,
-                lineHeight: 16
-            )
-
-        case .caption2:
-            return baseAttributes(
-                sizeProvider: 11,
-                font: VitaminFontFamily.Roboto.regular,
-                lineHeight: 13
-            )
-
-        case .button:
-            return baseAttributes(
-                sizeProvider: 16,
-                font: VitaminFontFamily.Roboto.bold,
-                lineHeight: 16
-            )
-        }
+        let adjustedFont = scaledFont
+        let adjustedSize = adjustedFont.pointSize
+        let adjustedLineHeight = scaledLineHeight(for: adjustedFont)
+        let paragraphStyle = paragraphStyle(lineHeight: adjustedLineHeight)
+        let baselineOffset = baselineOffset(lineHeight: adjustedLineHeight, size: adjustedSize)
+        return [
+            .foregroundColor: VitaminColor.Core.Content.primary,
+            .font: adjustedFont,
+            .paragraphStyle: paragraphStyle,
+            .baselineOffset: baselineOffset
+        ]
     }
 }
 
+// MARK: - Public String extension
+
 extension String {
+    /// Get the `NSAttributedString` to render provided style.
+    /// - Parameter textStyle: The style that we want to apply.
+    /// - Returns: A `NSAttributedString` with all attributes.
     public func styled(as textStyle: VitaminTextStyle) -> NSAttributedString {
         NSAttributedString(string: self, attributes: textStyle.attributes)
     }
 
+    /// Get the `NSAttributedString` to render provided style.
+    /// - Parameters:
+    ///   - textStyle: The style that we want to apply.
+    ///   - color: The color for the text. Default: Black.
```

Comment:
The indication of the default color seems strange here, since there is no default value on the color. It's just that if you do not specify a color, you fallback to the method above.
Maybe we could merge the two methods, by providing a real default value.

Furthermore, if I make no mistake, even if we keep both methods, the default value is not Black, but Content.primary

Reply:
I'm not sure why I didn't do like that.
Fixed 👍

Reply:
To be honest, I think I am the one to blame, because I am the one who added the method with a `color` parameter two months ago ;)